core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushPredicateIntoTableScan.java

```diff
@@ -275,6 +285,23 @@ private boolean arePlansSame(FilterNode filter, TableScanNode tableScan, PlanNod
         return Optional.of(tableScan);
     }
 
+    // PushProjectionIntoTableScan is executed after AddExchanges and UseConnectorTableScanPartitioning.
```

Comment:
Connector could return different table handle after `PushProjectionIntoTableScan` with different partitioning. Since we planned with specific partitioning (`AddExchanges`) subsequent `PushProjectionIntoTableScan` should not change that partitioning